core/src/main/java/org/eclipse/krazo/core/ViewResponseFilter.java

```diff
@@ -142,16 +145,13 @@ public void filter(ContainerRequestContext requestContext,
         Object entity = responseContext.getEntity();
         final Class<?> entityType = entity != null ? entity.getClass() : null;
         if (entityType == null) {       // NO_CONTENT
-            View an = getAnnotation(method, View.class);
-            if (an == null) {
-                an = getAnnotation(method.getDeclaringClass(), View.class);
-            }
-            if (an != null) {
+            String viewPath = viewPathResolver.pathFor(method);
```

Comment:
does not change behaviour, just give ability to override how view  name is resolved in case of void for example

`
public void myview(){
    this.models.put("a","a");
}
//will force to annotate with @View("my.jsp") if no alternative impl is provided,
//as  alternative you can change the behaviour delegating to another impl resolve view name for you with no need of annotation in case of void. 

`
Both cases you are always have a view name resolved by default spec annotation or by another point of extension

Reply:
> as alternative you can change the resolver delegating to another impl resolve view name for you with no need of annotation in case of void. 

Yeah and that's the point where I think such extensions can / will violate the spec. Regarding `void` methods there is declared that
> A controller method that returns void is REQUIRED to be decorated by `@View`.

And if we implement extension points this way, I think we need to consider the other return types (`String` and `Response`) too, as someone may want to change their default behavior too.

Reply:
> A controller method that returns void is REQUIRED to be decorated by @View
Maybe change specs to: A controller method that returns void MUST provide a view name resolution. Can be decorated by @View or implement a ViewPathResolver. 

Maybe spec can be less restrictive in future. 

> And if we implement extension points this way, I think we need to consider the other return types (String and Response) too, as someone may want to change their default behavior too.

I agree.